cartography/intel/opal/opal_common.py

```diff
@@ -0,0 +1,134 @@
+        if okta_group_id:
+            auto_approved_access.append({
+                'resource_id': resource_id,
+                'okta_group_id': okta_group_id,
+            })
+    return auto_approved_access
+
+
+def parse_opal_resource_access_configuration(
+        opal_resources: List[Dict[str, Any]],
+        opal_to_okta_map: Dict[str, str],
+        opal_admin_to_okta_map: Dict[str, str],
+) -> Tuple[List[Dict[str, Any]], List[Dict[str, Any]]]:
+    """
+    Return auto approved access in an OpalResource that is tied back to the OktaGroup that has autoapproved access.
+    """
+    logger.info(f"Getting auto approved and manual approved access for {len(opal_resources)} opal resources")
+    auto_approved_access = []
+    manual_approved_access = []
+    for resource in opal_resources:
+        resource_id: str = str(resource.get('resource_id', ""))
+        for request_config in resource.get('request_configurations', []):
+            if request_config.get('auto_approval'):
+                auto_approved_access.extend(
+                    get_auto_approved_access(
+                        request_config, opal_to_okta_map, resource_id,
+                    ),
+                )
+            else:
+                # If there is no auto approval, then there is manual approval.
+                manual_approved_access.extend(
+                    get_manual_approved_access(
+                        request_config, opal_admin_to_okta_map, resource_id,
+                    ),
+                )
+    return auto_approved_access, manual_approved_access
+
+
+def load_auto_approved_access(neo4j_session: Any, update_tag: int, auto_approved_access: List[Dict[str, Any]]) -> None:
+    """
+    Load the auto approved access relationships into the graph.
+    """
+    logger.info(f"Loading {len(auto_approved_access)} auto approved access relationships")
+    neo4j_session.run(
+        """
+        UNWIND $auto_approved_access AS access
+        MATCH (opal_resource:OpalResource {id: access.resource_id})
+        MATCH (okta_group:OktaGroup {id: access.okta_group_id})
+        MERGE (okta_group)-[r:CAN_AUTO_APPROVE_ACCESS]->(opal_resource)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = $update_tag
+        """,
+        auto_approved_access=auto_approved_access, update_tag=update_tag,
+    )
+
+
+def load_manual_approvers(neo4j_session: Any, update_tag: int, manual_approved_access: List[Dict[str, Any]]) -> None:
+    """
+    Load the mapping of OktaGroup to OpalResources they can manually approve access for into Neo4j.
+    """
+    logger.info(f"Loading {len(manual_approved_access)} manual approved access relationships")
+    neo4j_session.run(
+        """
+        UNWIND $manual_approved_access AS access
+        MATCH (g:OktaGroup {id: access.okta_group_id})
+        MATCH (r:OpalResource {id: access.resource_id})
+        MERGE (g)-[rel:CAN_MANUALLY_APPROVE_ACCESS]->(r)
```

Comment:
Same comment here about making this a rel schema defined on the OpalResource model and then using a read_transaction function to get a list of the IDs you need.